core/blockly.js

```diff
@@ -230,23 +226,14 @@ Blockly.onContextMenu_ = function(e) {
   }
 };
 
-/**
- * Close tooltips, context menus, dropdown selections, etc.
- * @param {boolean=} opt_onlyClosePopups Whether only popups should be closed.
- */
-Blockly.hideChaff = function(opt_onlyClosePopups) {
-  Blockly.Tooltip.hide();
-  Blockly.WidgetDiv.hide();
-  Blockly.DropDownDiv.hideWithoutAnimation();
-
-  var onlyClosePopups = !!opt_onlyClosePopups;
-  var workspace = Blockly.common.getMainWorkspace();
-  var autoHideables = workspace.getComponentManager().getComponents(
-      Blockly.ComponentManager.Capability.AUTOHIDEABLE, true);
-  autoHideables.forEach(function(autoHideable) {
-    autoHideable.autoHide(onlyClosePopups);
-  });
-};
+// Add a getter for Blockly.hideChaff, for legacy reasons.
+Object.defineProperty(Blockly, 'hideChaff', {
+  get: function() {
+    Blockly.utils.deprecation.warn(
+        'Blockly.hideChaff', 'September 2021', 'September 2022');
+    return Blockly.common.getMainWorkspace().hideChaff;
+  }
+});
```

Comment:
Why use an accessor here—wouldn't a stub wrapper suffice?:

```suggestion
/**
 * Close tooltips, context menus, dropdown selections, etc.
 * @deprecated Use Blockly.common.getMainWorkspace().hideChaff()
 * @param {boolean=} opt_onlyClosePopups Whether only popups should be closed.
 */
Blockly.hideChaff = function(opt_onlyClosePopups) {
  Blockly.utils.deprecation.warn(
      'Blockly.hideChaff', 'September 2021', 'September 2022');
  Blockly.common.getMainWorkspace().hideChaff(opt_onlyClosePopups);
};
```

This gives a place to hang the `@deprecation` notice and preserves type information.